master/cmd/determined-gotmpl/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"text/template"
+
+	"github.com/ghodss/yaml"
+
+	"github.com/Masterminds/sprig/v3"
+	"github.com/huandu/xstrings"
+)
+
+func stdinData() map[string]interface{} {
+	stat, _ := os.Stdin.Stat()
+	if (stat.Mode() & os.ModeCharDevice) != 0 {
+		return nil
+	}
+	stdin, err := ioutil.ReadAll(os.Stdin)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var data = map[string]interface{}{}
+	err = yaml.Unmarshal(stdin, &data)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return data
+}
+
+func toYaml(v interface{}) (string, error) {
+	data, err := yaml.Marshal(v)
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+// RunTmpl renders template `files` given context `data`.
+func RunTmpl(data map[string]interface{}, files []string) *bytes.Buffer {
+	funcs := sprig.TxtFuncMap()
+	funcs["cwd"] = os.Getwd
+	funcs["args"] = func() []string { return os.Args }
+	funcs["templateFiles"] = func() []string { return files }
+	funcs["absPath"] = filepath.Abs
+	funcs["upperFirst"] = xstrings.FirstRuneToUpper
+	funcs["lowerFirst"] = xstrings.FirstRuneToLower
+	funcs["reflectKind"] = func(val interface{}) string {
+		if val == nil {
+			return ""
+		}
+		return reflect.TypeOf(val).Kind().String()
+	}
+	funcs["toYaml"] = toYaml
+
+	tmpl, err := template.
+		New(filepath.Base(files[0])).
+		Funcs(funcs).
+		ParseFiles(files...)
+	if err != nil {
+		log.Fatal(err)
+	}
+	b := new(bytes.Buffer)
+	err = tmpl.Execute(b, data)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return b
+}
+
+func main() {
+	flag.Usage = func() {
+		fmt.Fprint(flag.CommandLine.Output(),
+			`Usage: determined-gotmpl [-i data.yaml] [-o output.txt] [key=value] template [template ...]
+
+Based on https://github.com/NateScarlet/gotmpl
```

Comment:
why not just use gotmpl, as opposed to this binary? i guess there is some customization in here, what does it not support?

Reply:
yaml input and `toYaml` helper, mostly. added a comment.